src/form-builder/SSNField.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import { FieldHookConfig, useField } from 'formik';
+
+import { FieldProps } from './types';
+import { chainValidations, isValidSSN, required } from '../utils/validation';
+import { VaTextInput } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+export type SSNProps = FieldProps<string>;
+
+/**
+ * Renders the SSNField component
+ *
+ * @beta
+ */
+const SSNField = (props: SSNProps): JSX.Element => {
+  // Note: In this component, the Formik variable "field.value" holds the raw SSN value,
+  // while the useState variable "ssn" controls the view and will render the masked SSN to the page
+  const withValidation = {
+    ...props,
+    validate: chainValidations(props, [required, isValidSSN]),
+  };
+  const [field, meta, helpers] = useField(
+    withValidation as FieldHookConfig<string>
+  );
+  const id = props.id || props.name;
+
+  const [ssn, setSSN] = useState('');
+
+  const onFocus = () => {
+    if (!field.value) return;
+    const valueWithoutDashes = field.value.replaceAll('-', '');
+
+    setSSN(valueWithoutDashes);
+  };
+
+  const onBlur = (event: any) => {
+    const { value } = event.target;
+    const ssnString: string = value.replaceAll('-', '');
+    helpers.setValue(ssnString);
+    helpers.setTouched(true);
+
+    let maskedSSNString = '';
```

Comment:
Could we check with our accessibility experts on masking before we commit this to the forms library? There have been a couple of conversations about masking in the accessibility channel recently, and it seems like masking can decrease accessibility. Here is one: https://adhoc.slack.com/archives/C6NT0TTU3/p1650572355414659